src/main/kotlin/de/bigboot/ggtools/fang/service/MistforgeService.kt

```diff
@@ -90,21 +112,37 @@ class MistforgeService: AutostartService, KoinComponent {
                 4 -> 4
                 else -> 5
             }
-
+            
             val icon = ImmutableImage.loader()
-                .fromBytes(mistforge.getHeroImage(guide.heroId, "${name}.png").bytes())
+                .fromBytes(mistforge.getHeroImage(heroId, "${name}.png").bytes())
                 .scaleTo(96, 96, ScaleMethod.Bicubic)
 
             img = img.overlay(icon, x, y)
         }
-
+        
+        var talentId = 0
```

Comment:
use `when` instead of this if/else thing, also rather return without parsing the url than setting some nonsense default

Reply:
Also I am setting a nonsense default because doing so will make it "backwards" compatible with mistforge. If you go to [https://mistforge.net/build_viewer?b](https://mistforge.net/build_viewer?b) it has some weird defaults, so I want those to be in the bot too.

Reply:
This is just a bug, the actual builder won't let you finish without selecting all skills, you can try: https://mistforge.net/hero_builder